subsys/nfc/tnep/tag.c

```diff
@@ -0,0 +1,701 @@
+
+const struct tnep_state tnep_state_machine[] = {
+	{
+		TNEP_STATE_DISABLED,
+		tnep_sm_disabled
+	},
+	{
+		TNEP_STATE_SERVICE_READY,
+		tnep_sm_service_ready
+	},
+	{
+		TNEP_STATE_SERVICE_SELECTED,
+		tnep_sm_service_selected
+	},
+};
+
+static void tnep_tx_msg_clear(void)
+{
+	nfc_ndef_msg_clear(tnep.msg);
+}
+
+static int tnep_tx_msg_encode(void)
+{
+	int err = 0;
+	unsigned int key;
+	size_t len;
+
+
+	if (tnep.current_buff == tnep.tx.data) {
+		tnep.current_buff = tnep.tx.swap_data;
+	} else {
+		tnep.current_buff = tnep.tx.data;
+	}
+
+	memset(tnep.current_buff, 0, tnep.tx.len);
+
+	len = tnep.tx.len;
+
+	if (tnep.msg->record_count > 0) {
+		err = nfc_ndef_msg_encode(tnep.msg,
+					  tnep.current_buff,
+					  &len);
+	}
+
+	key = irq_lock();
+
+	__ASSERT_NO_MSG(tnep.data_set);
+
+	tnep.data_set(tnep.current_buff, tnep.tx.len);
+
+	irq_unlock(key);
+
+	return err;
+}
+
+static int tnep_tx_msg_add_rec(const struct nfc_ndef_record_desc *record)
+{
+	int err = 0;
+
+	if (!record) {
+		return -EINVAL;
+	}
+
+	err = nfc_ndef_msg_record_add(tnep.msg,
+				      record);
+
+	return err;
+}
+
+static int tnep_tx_msg_add_rec_status(enum nfc_tnep_status_value status)
+{
+	int err = 0;
+
+	if (status == NFC_TNEP_STATUS_PROTOCOL_ERROR) {
+		tnep_tx_msg_clear();
+	}
+
+	status_record.status = status;
+
+	err = tnep_tx_msg_add_rec(&NFC_NDEF_TNEP_RECORD_DESC(status_record));
+
+	return err;
+}
+
+static int tnep_tx_initial_msg_set(void)
+{
+	int err = 0;
+
+	tnep_tx_msg_clear();
+
+	for (size_t i = 0; i < tnep.svc_cnt; i++) {
+		err = tnep_tx_msg_add_rec(tnep.svc[i].ndef_record);
+		if (err) {
+			return err;
+		}
+	}
+
+	if (!tnep.records) {
+		return tnep_tx_msg_encode();
+	}
+
+	for (size_t i = 0; i < tnep.records_cnt; i++) {
+		err = tnep_tx_msg_add_rec(&tnep.records[i]);
+		if (err) {
+			return err;
+		}
+	}
+
+	return tnep_tx_msg_encode();
+}
+
+static bool ndef_check_rec_type(const struct nfc_ndef_record_desc *record,
+				const u8_t *type_field,
+				const u32_t type_field_length)
+{
+	int cmp_res;
+
+	if (record->type_length != type_field_length) {
+		return false;
+	}
+
+	cmp_res = memcmp(record->type, type_field, type_field_length);
+
+	return (cmp_res == 0);
+}
+
+static int tnep_svc_set_active(u8_t *uri_name, size_t uri_length)
+{
+
+	for (size_t i = 0; i < tnep.svc_cnt; i++) {
+		if ((tnep.svc[i].parameters->uri_length == uri_length) &&
+		    !memcmp(uri_name,
+			    tnep.svc[i].parameters->uri, uri_length)) {
+			tnep.svc_active = &tnep.svc[i];
+
+			return TNEP_SVC_SELECTED;
+		}
+	}
+
+	tnep.svc_active = NULL;
+
+	return TNEP_SVC_DESELECTED;
+}
+
+static int tnep_svc_select_from_msg(void)
+{
+	int err;
+	enum tnep_svc_record_status svc_status = TNEP_SVC_NOT_FOUND;
+	const struct nfc_ndef_msg_desc *msg_p;
+	u8_t desc_buf[NFC_NDEF_PARSER_REQIRED_MEMO_SIZE_CALC(
+				CONFIG_NFC_TNEP_RX_MAX_RECORD_CNT)];
+	size_t desc_buf_len = sizeof(desc_buf);
+
+	/* Parse rx buffer to NDEF message */
```

Comment:
this comment is not very useful - `nfc_ndef_msg_parse` name is descriptive enough

Reply:
excessive comments only pollute the code

Reply:
Ok. Removed